.github/workflows/build.yml

```diff
@@ -710,16 +711,23 @@ jobs:
         if: ${{ startsWith(github.ref, 'refs/tags/v') }}
         run: mv uarch-logs/uarch-riscv-tests-json-logs.tar.gz uarch-logs/uarch-riscv-tests-json-logs-v${{ env.MACHINE_EMULATOR_VERSION }}.tar.gz
 
-      - name: Upload TAR and DEB to GitHub Release
+      - name: Create generated files patch
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: |
+          mv artifacts/machine-c-version.h src
+          mv artifacts/uarch-pristine-ram.c uarch
+          mv artifacts/uarch-pristine-hash.c uarch
+          make create-generated-files-patch
+
+      - name: Upload products to GitHub Release
         uses: softprops/action-gh-release@v1
         if: ${{ startsWith(github.ref, 'refs/tags/v') }}
         with:
           draft: true
           files: |
             artifacts/cartesi-machine-*.deb
             artifacts/uarch-ram.bin
-            artifacts/uarch-pristine-ram.c
-            artifacts/uarch-pristine-hash.c
+            add-generated-files.diff
```

Comment:
I think it is way common to name `.patch` instead of `.diff`, here is an example for aports (Alpine package tree):

```
bart@archawk:~/repos/aports (ltrace-riscv64 %) $ find -name '*.patch' | wc
   5171    5171  237831
bart@archawk:~/repos/aports (ltrace-riscv64 %) $ find -name '*.diff' | wc
      6       6     295
```

So I suggest following the standard.